### ISSUE TITLE
INFO Exception in queued task: Error: No replica set primary available for query with ReadPreference PRIMARY

### ISSUE THREAD

**Issue by stephentcannon** [meteor/meteor#1060]
Seeing this a lot on apps that have been migrated to production-db-a3.meteor.io:27017/atomic_etherpos_com

[Mon May 20 2013 16:46:30 GMT+0000 (UTC)] INFO Exception in queued task: Error: No replica set primary available for query with ReadPreference PRIMARY
    at Object.Future.wait (/meteor/dev_bundles/0.3.0/lib/node_modules/fibers/future.js:319:16)
    at _.extend._nextObject (app/packages/mongo-livedata/mongo_driver.js:466:47)
    at _.extend.forEach (app/packages/mongo-livedata/mongo_driver.js:487:22)
    at _.extend.getRawObjects (app/packages/mongo-livedata/mongo_driver.js:531:12)
    at _.extend._pollMongo (app/packages/mongo-livedata/mongo_driver.js:802:46)
    at Object._.extend._unthrottledEnsurePollIsScheduled [as task](app/packages/mongo-livedata/mongo_driver.js:746:12)
    at _.extend._run (app/packages/meteor/fiber_helpers.js:124:18)
    at _.extend._scheduleRun (app/packages/meteor/fiber_helpers.js:102:14)
    - - - - -
    at ReplSet.checkoutReader (/meteor/containers/003e5cf7-b661-9001-1f2c-0e5139f21cdb/bundle/app/packages/mongo-livedata/node_modules/mongodb/lib/mongodb/connection/repl_set.js:1099:14)
    at Cursor.nextObject (/meteor/containers/003e5cf7-b661-9001-1f2c-0e5139f21cdb/bundle/app/packages/mongo-livedata/node_modules/mongodb/lib/mongodb/cursor.js:638:108)
    at Future.wrap [as _synchronousNextObject](/meteor/dev_bundles/0.3.0/lib/node_modules/fibers/future.js:30:6)
    at _.extend._nextObject (app/packages/mongo-livedata/mongo_driver.js:466:22)
    at _.extend.forEach (app/packages/mongo-livedata/mongo_driver.js:487:22)
    at _.extend.getRawObjects (app/packages/mongo-livedata/mongo_driver.js:531:12)
    at _.extend._pollMongo (app/packages/mongo-livedata/mongo_driver.js:802:46)
    at Object._.extend._unthrottledEnsurePollIsScheduled [as task](app/packages/mongo-livedata/mongo_driver.js:746:12)
    at _.extend._run (app/packages/meteor/fiber_helpers.js:124:18)
    at _.extend._scheduleRun (app/packages/meteor/fiber_helpers.js:102:14)

**Comment by TimothyAhong** [comment]
I'm running into this issue again when I attempt to deploy via meteor deploy x.meteor.com. I have posted more details on stackoverflow: http://stackoverflow.com/questions/30063102/meteorjs-mongodb-deploy-error

**Comment by nabiltntn** [comment]
It's back again, i have the issue after redeploying my app, the same error in logs :No replica set primary available for query with ReadPreference PRIMARY.
The url of my application : http://www.tuniliv.com

Thanks.

**Comment by RazZziel** [comment]
My site (parkapp.meteor.com) is still offline since yesterday evening, even after a re-deploy: http://pastebin.com/raw.php?i=UY6z9Bqr
I can't even extract the mongo url with `meteor mongo --url {site}`: http://pastebin.com/raw.php?i=cSgYLmLQ

**Comment by jirkamelichar** [comment]
I'm having exactly the same issue as well when deploying to ```*.meteor.com```

Log: http://pastebin.com/GpWf8i6f

Connection to Mongo via ```meteor mongo --url {site}``` not working either.

**Comment by cpalanzo** [comment]
Also having this problem since this morning after deploy.

**Comment by RazZziel** [comment]
Now it's working again for me

**Comment by nabiltntn** [comment]
Me too, things are getting better !

**Comment by jirkamelichar** [comment]
Working now - thank you!

**Comment by glasser** [comment]
We are aware of this outage to some of our Mongo servers and are actively working on resolving it.

**Comment by glasser** [comment]
(Let's track this in #4373.)

**Comment by theartsnetwork** [comment]
I am facing the same issue with my application site buzzin.artfrontierproject.com since night before yesterday. I couldn't connect with mongoBD and error page showing "This site is loading" message. I am waiting for this issue to be resolved sooner. Please help.

**Comment by abhisheksoni27** [comment]
How can this be resolved?

**Comment by jaredtmartin** [comment]
I'm having this problem with returnvisits.meteor.com

**Comment by niranjans** [comment]
I am getting this too on my App deployed on Modulus. 

```
Exception while invoking method 'login' Error: No replica set primary available for query with ReadPreference PRIMARY
```

**Comment by mrcoles** [comment]
This is happening to me for the last couple days on http://instaslideshow.meteor.com/

**Comment by NikosEfthias** [comment]
i have the same problem

**Comment by motomux** [comment]
I got the same issue with Meteor 1.2.1.
Is this issue solved on Meteor 1.3?

**Comment by stubbegianni** [comment]
Some guy made a comment on a similar problem over her:
https://github.com/meteor/meteor/issues/2092#issuecomment-158926195

Could this be the solution? I got this error yesterday but I don't know why exactly as the only thing that has changed is that I converted a standalone to a single replica set member for Meteor Oplog
